### ISSUE TITLE
Dynamic Linking does not work on Windows MinGW Target

### ISSUE THREAD

**Issue by AeroSW** [ziglang/zig#21103]
### Zig Version

0.13.0

## Setup
1. Download [raylib release version 5 for MinGW](https://github.com/raysan5/raylib/releases/download/5.0/raylib-5.0_win64_mingw-w64.zip)
2. Pull its contents into a directory local with the project, `./raylib`
3. Update build target to be MinGW (Windows + GNU = MinGW):
```
    const target = b.standardTargetOptions(.{
        .default_target = .{
            .os_tag = .windows,
            .cpu_arch = .x86_64,
            .abi = .gnu
        }
    });
```
4. Ensure you have your dependencies for `raylib` linked
```
    exe.linkLibC();
    exe.addIncludePath(b.path("./raylib/include"));
    exe.addLibraryPath(b.path("./raylib/lib"));
    exe.linkSystemLibrary2("uuid", .{
        .preferred_link_mode = .static,
        .needed = true
    });
    exe.linkSystemLibrary2("gdi32", .{
        .preferred_link_mode = .static,
        .needed = true
    });
    exe.linkSystemLibrary2("winmm", .{
        .preferred_link_mode = .static,
        .needed = true
    });
```
5. Attempt to have `build.zig` link `raylib` dynamically
```
    exe.linkSystemLibrary2("raylib", .{
        .preferred_link_mode = .dynamic,
        .needed = true,
    });
```
6. Attempt to run `zig build --verbose install`.  It should produce the following error:
```
error: error(link): DLL import library for -lraylib not found
error: DllImportLibraryNotFound
```

The attempting to search for the DLL seems to be tied to a `raylib.lib` file being present despite `MinGW` configuration using `libraylib.a` libraries.  This might have been an oversight when parsing configurations to search for DLLs.

### Expected Behavior

Expected behavior should be for Zig to be able to accurately find the DLL while using MinGW and not MSVC environment.

**Comment by dkareh** [comment]
Is the problem as simple as Zig looking for `raylib.lib` and `libraylib.dll.a` but not `libraylibdll.a` (as the import library is named in the linked release archive)? Relevant code: https://github.com/ziglang/zig/blob/bb70501060a8bfff/src/link/Coff/lld.zig#L486-L508

**Comment by AeroSW** [comment]
There are a few things that are weird about the code you pasted.  If Zig is default using GNU, then why is Zig by default using `.lib` in this code.  It seems counter intuitive.  `.lib` extension is related to MSVC compilation, not MinGW.  I think adding a branch here  makes sense where we also check for `lib{s}dll.a` should expand it.  I'm not sure how often software doesn't use a period to separate the library's name and `dll` for these static libs.

**Comment by AeroSW** [comment]
I just noticed another issue higher up in the code.  The `else` statement in the selected `if-else` block gets overridden on line 414.
https://github.com/ziglang/zig/blob/bb70501060a8bfff/src/link/Coff/lld.zig#L403-L414

**Comment by AeroSW** [comment]
What you recommended works for both `raylib.dll` and `libraylibdll.a`.  I still find it awkward how the zig source is set up for detecting DLLs using its `named` methods for linking.
###Code Below:
`libraylibdll.a`
```
    exe.linkLibC();
    exe.addIncludePath(b.path("./raylib/include"));
    exe.addLibraryPath(b.path("./raylib/lib"));
    exe.linkSystemLibrary2("uuid", .{
        .preferred_link_mode = .dynamic,
        .needed = true
    });
    exe.linkSystemLibrary2("gdi32", .{
        .preferred_link_mode = .dynamic,
        .needed = true
    });
    exe.linkSystemLibrary2("winmm", .{
        .preferred_link_mode = .dynamic,
        .needed = true
    });
    exe.addObjectFile(b.path("raylib/lib/libraylibdll.a"));
```
`raylib.dll`
```
    exe.linkLibC();
    exe.addIncludePath(b.path("./raylib/include"));
    exe.addLibraryPath(b.path("./raylib/lib"));
    exe.linkSystemLibrary2("uuid", .{
        .preferred_link_mode = .dynamic,
        .needed = true
    });
    exe.linkSystemLibrary2("gdi32", .{
        .preferred_link_mode = .dynamic,
        .needed = true
    });
    exe.linkSystemLibrary2("winmm", .{
        .preferred_link_mode = .dynamic,
        .needed = true
    });
    exe.addObjectFile(b.path("raylib/lib/raylib.dll"));
```

**Comment by dkareh** [comment]
Agreed, except the third parameter of `findLib` is the list of library search paths, so it wouldn't work for testing two suffixes. Anyway, adding a check for `lib{s}dll.a` seems like the most correct change. Hopefully there aren't any other conventions...

You mentioned that my workaround doesn't let the linker search for the library. Maybe we could provide some way to make the linker search for a library without adding any prefixes or suffixes. `ld`, for example, interprets [`-l:<filename>`](https://sourceware.org/binutils/docs/ld/Options.html#index-_002dl-namespec) as a command to search for a file called exactly `<filename>`. In fact, I think [Zig already supports the colon prefix](https://github.com/ziglang/zig/blob/bb70501060a8bfff/src/main.zig#L1887), but it expects the full path to the library instead of just the basename. But, I can't tell if this is a good idea or not.

**Comment by AeroSW** [comment]
Yeah, I realized that and forgot to edit my message.  Thank you for pointing that out as well!
